### ISSUE TITLE
Add search criteria to SearchableComboBox

### ISSUE THREAD

**Issue by abhinayagarwal** [controlsfx/controlsfx#1360]
`SearchableComboBox` currently filters the elements from user input using `String#contains` method.

The control can benefit if a pluggable search criteria can be added to the control. For example, the search uses `String#beginsWith` instead.

The new property can be a `BiFunction<String, String, Boolean>`, where the criteria can be defined as:

```
searchCriteria = (String v, String s) -> {
    return v.startsWith(s);
}
```

`v` is the string from the ComboBox's elements after the StringConverter is applied to it.

**Comment by danielpeintner** [comment]
> The new property can be a `BiFunction<String, String, Boolean>`,

Makes sense and aligns well with [`TableFilter`](https://github.com/controlsfx/controlsfx/blob/839a39aa6a75b8938256101918292a0044137fe7/controlsfx/src/main/java/org/controlsfx/control/table/TableFilter.java#L111)

**Comment by delvh** [comment]
> The new property can be a `BiFunction<String, String, Boolean>`, where the criteria can be defined as:

I think a `BiPredicate<String,String>` is more fitting as it already returns a boolean…

**Comment by james-d** [comment]
Surely it should be a `BiPredicate<T, String>`?

**Comment by willhartung** [comment]
If you're going to dig into this, then simply exposing (protected/public) "createFilteredList" so it can be replaced by a subclass would be useful because then it's no longer bound to a filter on a list of strings (or whatever), but someone could potentially connect it to a db. 

Or, perhaps, also open up "updateFilter", so we can add timer logic to run the query in the background. But all of those are private, so we can't do anything.

The predicate ideas are fine, but exposing the entire mechanism can be useful as well.
